test/integration/consumer_test.go

```diff
@@ -149,6 +149,23 @@ func TestConsumerCredential(t *testing.T) {
 		return resp.StatusCode == http.StatusOK
 	}, ingressWait, waitTick)
 
+	t.Logf("updating credential to confirm references update store copy")
+	credential, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Get(ctx, credential.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+	credential.StringData = map[string]string{"username": "new_consumer_credential"}
+	_, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Update(ctx, credential, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	assert.Eventually(t, func() bool {
+		req := helpers.MustHTTPRequest(t, "GET", proxyURL, "/test_consumer_credential", nil)
+		req.SetBasicAuth("new_consumer_credential", "test_consumer_credential")
+		resp, err := helpers.DefaultHTTPClient().Do(req)
+		if err != nil {
+			return false
+		}
+		defer resp.Body.Close()
+		return resp.StatusCode == http.StatusOK
+	}, ingressWait, waitTick)
```

Comment:
Could we cover this in an `envtest` instead?

Reply:
I don't think so. There's no visible effect on Kubernetes resources, and AFAIK envtest needs that to be able to cover something.

Do we have any existing envtests that can inspect parser output? I'm not aware of any. I think you could maybe do it if you created a special config apply mode that ignores the Kong admin API entirely and just reports the output at the [config export endpoint](https://docs.konghq.com/kubernetes-ingress-controller/latest/troubleshooting/#dumping-generated-kong-configuration) regardless.

You could then have the test scrape that and walk an unmarshaled YAML object to check if some value matches an expected one. Kinda like golden tests but you can check config state across resource updates.

Reply:
@pmalek added one recently that inspects the config: https://github.com/Kong/kubernetes-ingress-controller/pull/4786

Reply:
Gotcha. Doing so would probably mean porting the whole of this test to that approach though, and the main thrust of this is to make the updates stick. I think we can move forward with that and port the test elsewhere.